example/reduceRGBbySIMD/src/reduceRGBbySIMD.cpp

```diff
@@ -0,0 +1,324 @@
+        std::uint32_t* grayscaleRGBA,
+        size_t chunkSize) const
+    {
+        const size_t globalIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0];
+
+        // Process the SAME memory pattern as SIMD: multiple elements per thread
+        size_t startIdx = globalIdx * chunkSize;
+
+        // Process 'chunkSize' packed pixels sequentially (same as SIMD width)
+        for(size_t i = 0; i < chunkSize; ++i)
+        {
+            size_t idx = startIdx + i;
+            std::uint32_t packed = packedRGB[idx];
+
+            // SAME bit operations as SIMD, but scalar
+            // Format: 0xAARRGGBB (following Matthias Kretz's CppCon demo)
+            std::uint32_t a = packed >> 24; // Alpha channel
+            std::uint32_t r = (packed >> 16) & 0xFF; // Red channel
+            std::uint32_t g = (packed >> 8) & 0xFF; // Green channel
+            std::uint32_t b = packed & 0xFF; // Blue channel
+
+            // SAME integer arithmetic as SIMD (standard luminance formula)
+            // Alpha channel is extracted but NOT used in grayscale calculation (following Matthias Kretz's CppCon
+            // demo)
+            std::uint32_t gray = (r * intScalarR + g * intScalarG + b * intScalarB) >> 8;
+
+            // Pack grayscale result with preserved alpha: 0xAARRGGBB -> 0xAAGGGGGG
+            // Grayscale value replaces RGB, alpha is preserved
+            std::uint32_t result = (a << 24) | (gray << 16) | (gray << 8) | gray;
+            grayscaleRGBA[idx] = result;
+        }
+    }
+};
+
+template<typename Acc>
+void runPackedRGBAComparison(
+    alpaka::Dev<Acc> const& devAcc,
+    alpaka::Queue<Acc, alpaka::Blocking>& queue,
+    std::vector<std::uint32_t> const& packedRGB,
+    std::vector<std::uint32_t>& grayscaleResult,
+    size_t elements)
+{
+    using Dim = alpaka::DimInt<1u>;
+    using Idx = std::size_t;
+
+    // Create SIMD type to get width
+    using SimdType = alpaka::simd::PortableSimd<std::uint32_t, Acc>;
+    SimdType temp;
+    size_t simdWidth = temp.size();
+
+    std::cout << "simdWidth for packed RGBA (uint32_t) is " << simdWidth << std::endl;
+
+    // Calculate reduced thread count for fair comparison
+    size_t workDiv = (elements + simdWidth - 1) / simdWidth; // Round up
+    auto const workDivPacked = alpaka::WorkDivMembers<Dim, Idx>{Idx(workDiv), Idx(1), Idx(1)};
+
+    // Allocate device memory
+    auto devicePackedRGB = alpaka::allocBuf<std::uint32_t, Idx>(devAcc, elements);
+    auto deviceGrayscale = alpaka::allocBuf<std::uint32_t, Idx>(devAcc, elements);
+
+    // Copy input data
+    alpaka::memcpy(queue, devicePackedRGB, packedRGB);
+
+    float scalarPackedTime = 0.0f;
+    float simdPackedTime = 0.0f;
+
+    // Test 1: Scalar Packed RGBA (FAIR COMPARISON - same data format, same algorithm)
+    {
+        std::cout << "Scalar: " << workDiv << " threads, " << simdWidth << " elements each" << std::endl;
+
+        auto start = std::chrono::high_resolution_clock::now();
```

Comment:
`wait()` is missing it could be you messure the memcopy if you use a sync queue.